crates/starknet-devnet/tests/test_estimate_fee.rs

```diff
@@ -676,4 +678,202 @@ mod estimate_fee_tests {
+    }
+
+    #[tokio::test]
+    async fn estimate_fee_of_multiple_failing_txs_should_return_index_of_the_first_failing_transaction()
+     {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start devnet");
+
+        // get account
+        let (signer, account_address) = devnet.get_first_predeployed_account().await;
+        let mut account = SingleOwnerAccount::new(
+            &devnet.json_rpc_client,
+            signer.clone(),
+            account_address,
+            devnet.json_rpc_client.chain_id().await.unwrap(),
+            ExecutionEncoding::New,
+        );
+
+        account.set_block_id(BlockId::Tag(BlockTag::Latest));
+
+        let (flattened_contract_artifact, casm_hash) =
+            get_flattened_sierra_contract_and_casm_hash(CAIRO_1_PANICKING_CONTRACT_SIERRA_PATH);
+        let class_hash = flattened_contract_artifact.class_hash();
+
+        let estimate_fee_resource_bounds = ResourceBoundsMapping {
+            l1_gas: ResourceBounds { max_amount: 0, max_price_per_unit: 0 },
+            l2_gas: ResourceBounds { max_amount: 0, max_price_per_unit: 0 },
+        };
+
+        // call non existent method in UDC
+        let calls = vec![Call {
+            to: UDC_CONTRACT_ADDRESS,
+            selector: get_selector_from_name("no_such_method").unwrap(),
+            calldata: vec![
+                class_hash,
+                Felt::from_hex_unchecked("0x123"), // salt
+                Felt::ZERO,
+                Felt::ZERO,
+            ],
+        }];
+
+        let calldata = account.encode_calls(&calls);
+
+        let is_query = true;
+        let nonce_data_availability_mode = DataAvailabilityMode::L1;
+        let fee_data_availability_mode = DataAvailabilityMode::L1;
+
+        let expected_error = devnet
+            .json_rpc_client
+            .estimate_fee(
+                [
+                    BroadcastedTransaction::Declare(
+                        starknet_rs_core::types::BroadcastedDeclareTransaction::V3(
+                            BroadcastedDeclareTransactionV3 {
+                                sender_address: account_address,
+                                compiled_class_hash: casm_hash,
+                                signature: vec![],
+                                nonce: Felt::ZERO,
+                                contract_class: Arc::new(flattened_contract_artifact.clone()),
+                                resource_bounds: estimate_fee_resource_bounds.clone(),
+                                tip: 0,
+                                paymaster_data: vec![],
+                                account_deployment_data: vec![],
+                                nonce_data_availability_mode,
+                                fee_data_availability_mode,
+                                is_query,
+                            },
+                        ),
+                    ),
+                    BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V3(
```

Comment:
Perhaps `broadcasted_invoke_v1_for_estimation` can be changed to use v3 and used here too? And potentially can be moved to the utils file, and used in other tests where `BroadcastedTransaction::Invoke` is used? If it would take too long, we don't have to deal with it in this PR, but a separate one.